src/navigation/failure_identification/watchdog.py

```diff
@@ -0,0 +1,11 @@
+import numpy as np
+import rospy
+from pandas import DataFrame
+
+
+class WatchDog:
+    def __init__(self, collector_in):
+        self.collector = collector_in
```

Comment:
why do you need this collector variable?

Reply:
This is also used when trying to detect if the rover is stuck, which is for another PR.